tuist/tuist
# Unifying manifest target generation

Part of: #205 

### Short description 📝

When generating projects, Tuist generates user defined targets as well as a manifest target to host the project manifest (`Project.swift` file). Currently the generator has a set of dedicated methods to generating the manifest target and its config. This could be simplified by re-using the existing target generation code for the other targets.

### Solution 📦

When creating the project models, we could add one more target (the manifest target) with the necessary custom settings. 

### Implementation 👩‍💻👨‍💻

- Introduce a manifest target generator that generates the necessary `Target` model for the manifest
- Add the manifest target to the list of project targets when loading its model

### Test Plan ⚒

- Verify unit tests pass via `swift test`
- Verify acceptance tests pass via `bundle exec rake features`
- Generate any fixture and validate the `Manifest` target is still generated
- Verify `Project.swift` is located within a top level "Manifest" group
- Verify autocomplete still work within the `Project.swift` file
